### DISCUSSION TITLE
Test gdpr consent without is_test_europe_user_consent

### DISCUSSION THREAD

**Discussion by Xioor** [Poing-Studios/Godot-AdMob-Android-iOS#71] [General]
I was wondering what would trigger a gdpr prompt in release? Is it using the users ip address, account geographic location information or something else?

**Comment by gumaciel** [comment]
Hey! 

Probably is by IP, we don't know for sure.

When we set `is_test_europe_user_consent = true`, we are doing this:

https://developers.google.com/admob/ump/android/quick-start#force_a_geography

In other words, we are telling to our application that we want to debug the Consent and we are on EEA.

**Reply by gumaciel** [reply]
@Xioor , A random question, what name do you think would be best to be able to call the admob_api (singleton) methods?

1. AdMob (I think this is the best, but it is not possible to use it due to a bug with Plugins, it would only work if I change the Android and iOS Plugin name)
2. MobileAds (it's the current name, comes from AdMob: https://developers.google.com/admob/android/quick-start#initialize_the_mobile_ads_sdk)
3. GoogleAdMob
4. AdMobAPI
5. Any other that you prefer and is not listed

**Reply by Xioor** [reply]
In the next few days, I will test if a EU IP if what triggers the gdpr prompt without is_test_europe_user_consent and let you know! As for your question on the naming for the singleton, I like MobileAds since that is the current one. But if you have to change it for some reason, I think GoogleAdMob is great.

**Reply by gumaciel** [reply]
Thank you!

**Reply by Xioor** [reply]
I tried a United Kingdom vpn ip and I do not get the prompt with is_test_europe_user_consent = false on Release or Debug. In a few days, I will test on a a few device that is actually located in Russia and Poland with this: https://developer.samsung.com/remotetestlab/galaxy/rtlDeviceList.action#502 Ill let you know how it went.

**Reply by gumaciel** [reply]
I think on Russia won't work, only on these:

[`Austria, Belgium, Bulgaria, Croatia, Republic of Cyprus, Czech Republic, Denmark, Estonia, Finland, France, Germany, Greece, Hungary, Ireland, Italy, Latvia, Lithuania, Luxembourg, Malta, Netherlands, Poland, Portugal, Romania, Slovakia, Slovenia, Spain and Sweden.`](https://www.gov.uk/eu-eea)

But thank you so much anyway!

**Reply by Xioor** [reply]
I tested a device in poland with https://developer.samsung.com/remotetestlab/galaxy/rtlDeviceList.action#502 and is_real = true with is_test_europe_user_consent = false as a release build and the gdpr prompt appeared on first launch! It did not appear again after the first launch, but this: https://github.com/Poing-Studios/Godot-AdMob-Android-iOS/issues/57 will remedy that. 
Good to know it's working!
